### ISSUE TITLE
Bildimport beim Ipad geht nicht > beendet scheinbar alles ordnungsgemäß, aber zeigt dann kein Bild.

### ISSUE THREAD

**Issue by McFly753** [Mojo1917/LocBase#248]

**Comment by Mojo1917** [comment]
Okay das ist ein ungewöhnlicher Fehler. Auf meinem Emulator bekomme ich den Fehler nicht erzeugt.
Ich denke dieser Fehler könnte auftreten, weil die Bilder möglicherweise nicht auf dem iPad liegen sondern nur in der iCloud. Es könnte dann hier zu einem Fehler beim Download kommen.

Um diesen Fehler eingrenzen zu können bräuchte ich zwei Infos von dir.
1. Werden die Fotos aus der iCloud geladen oder nur, wenn sie benötigt werden? Dies findest du unter iPad-Einstellungen > Fotos. Hier kann man die Option 'Speicher optimieren' oder 'Laden und Originale behalten' wählen. Bei 'Speicher optimieren' liegen sie nur in der iCloud und nur kleine 'Vorschaubilder' sind offline verfügbar.
2. Kannst du einmal ein Foto direkt mit der Kamera App aufnehmen und dieses dann im Anschluss in ein Objekt importieren? Dieses Bild liegt auf jeden Fall noch auf dem Gerät. Je nachdem wie sich die App verhält könnten wir den Fehler dann eingrenzen.

**Comment by McFly753** [comment]
Also: 
a) der Improt wird suaber abgeschlossen, wenn das Bild von der Kamera kommt
b) es wird nichts gemacht (Fehler von oben), wenn das Bild aus der Cloud kommt.
c) die App stürzt (lkurz nach dem Import, weil ich beim nächsten Statt das Bild in dem Objekt habe) ab, zum einen wenn das Bild vom Ipad selbst kommt (z.B. Screenshot oder auch das gemacht Bild von a)) oder aber auch, wenn in der Cloud ein Bild kommt, dass ich aber von jemanden anderen via Whats App bekommen habe (also keine Koordinaten).

**Comment by Mojo1917** [comment]
Super danke! Ich schaus mir an ;)

**Comment by Mojo1917** [comment]
Oh mein Kommentar von gerade war ein Fehler. Habe ihn jetzt doch gefunden. Behebe ihn jetzt ;)

**Comment by Mojo1917** [comment]
Okay der Download aus der Cloud sollte jetzt immer zuverlässig angeschlossen werden.

**Comment by McFly753** [comment]
Habe es dennoch getestet: es erfolgt kein Absturz mehr, aber Cloud-Bilder werden nicht importiert

**Comment by Mojo1917** [comment]
Okay. Mit dem nächsten Update sollte auch das zuverlässig funktionieren ;)

**Comment by McFly753** [comment]
Fast. // er macht nach den Cloud-Fotos irgendwie keinen refresh. Ich muss das Objket verlassen und dann (komischerweise 2x) reingehen, dass ich das hinzugefügte Bild auch angezeigt bekomme...

**Comment by Mojo1917** [comment]
Dies wird leider an der Verzögerung in der Synchronisation durch die iCloud liegen. Die Fotos sind dann zu dem Zeitpunkt noch nicht auf dem Gerät 'angekommen'. Sobald sie das jedoch sind sollte sie hier in diesem Dialog angezeigt werden.

**Comment by McFly753** [comment]
Verstehe. Ist aber didaktisch nicht gut. Ich würde dann gleich die Aktion nichmals machen und irgendwann habe ich das Bild öfters drin. Kannst Du abfragen, wann das oben ist und solange warten? Oder einen Lückenfüller mit "Download-Clowd"-Symbol anzeigen etc.?

**Comment by Mojo1917** [comment]
Nein das geht leider nicht. Man kann es sich so vorstellen, als würde das Gerät noch nicht 'wissen', dass es dieses Bild gibt. Sobald jedoch die Infos zur Existenz des Fotos übertragen ist greift unser Dialog hierauf zu und lädt die Bilder.

Da iOS hier von Apple so optimiert wurde, dass die Foto Mediathek sich möglichst Energieeffizient aktualisiert kann dies teilweise recht lange dauern. Sobald man die Fotos-App jedoch öffnet wird diese Synchronisation angestoßen und auch unsere App kann auf die Bilder zugreifen.

**Comment by McFly753** [comment]
Können wir dann nicht was anziegen (das Bild wird im HIntergrund geladen ...) - dass der User das versteht?

**Comment by Mojo1917** [comment]
Nein, leider wissen wir nicht, ob aktuell etwas geladen wird. Apple gewährt auf diese Daten keinen Zugriff.
Wir können einen solchen Dialog nur immer anzeigen. Dann aber auch, wenn keine Daten geladen werden.

**Comment by McFly753** [comment]
Lass uns hier bitte heute kurz telfoneiren, damit ich diese Logik verstehe.

**Comment by Mojo1917** [comment]
Klar gerne. Wann passt es bei dir?

**Comment by Mojo1917** [comment]
Okay wir haben jetzt eine ProgressBar und der Dialog verschwindet erst, wenn alles importiert ist ;)

Entschuldige das Missverständnis von meiner Seite.

**Comment by McFly753** [comment]
Danke! Kein Problem!
